tests/foreman/maintain/test_backup.py

```diff
@@ -0,0 +1,566 @@
+
+
+@pytest.mark.parametrize('backup_type', ['online', 'offline'])
+def test_positive_backup_restore_incremental(
+    sat_maintain, setup_backup_tests, module_synced_repos, backup_type
+):
+    """Incremental backup/restore end-to-end test.
+
+    :id: 77a66cea-8871-4f90-aba1-5295bfa95ecc
+
+    :steps:
+        1. take an initial backup
+        2. create additional content
+        3. take the incremental backup
+        4. check that appropriate files are created
+        5. restore initial backup and check system health
+        6. check that additional content is missing
+        7. restore incremental backup and check system health
+        8. check the initial and additional content was restored
+
+    :expectedresults:
+        1. initial backup succeeds
+        2. incremental backup succeeds
+        3. expected files are present in the backup
+        4. restore of all backups succeeds
+        5. system health check succeeds
+        6. content is present after restore
+    """
+    subdir = f'{BACKUP_DIR}backup-{gen_string("alpha")}'
+    result = sat_maintain.cli.Backup.run_backup(
+        backup_dir=subdir,
+        backup_type=backup_type,
+        options={'assumeyes': True},
+    )
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+    init_backup_dir = re.findall(fr'{subdir}\/satellite-backup-.*-[0-5][0-9]', result.stdout)[0]
+
+    # create additional content
+    secondary_repo = sat_maintain.api.Repository(
+        url=settings.repos.yum_3.url, product=module_synced_repos['custom'].product
+    ).create()
+    secondary_repo.sync()
+    secondary_repo = secondary_repo.read()
+
+    # take incremental backup
+    result = sat_maintain.cli.Backup.run_backup(
+        backup_dir=subdir,
+        backup_type=backup_type,
+        options={'assumeyes': True, 'incremental': init_backup_dir},
+    )
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+    # check for expected files
+    inc_backup_dir = re.findall(fr'{subdir}\/satellite-backup-.*-[0-5][0-9]', result.stdout)[0]
+    files = sat_maintain.execute(f'ls -a {inc_backup_dir}').stdout.split('\n')
+    files = [i for i in files if not re.compile(r'^\.*$').search(i)]
+
+    expected_files = OFFLINE_BACKUP_FILES if backup_type == 'offline' else ONLINE_BACKUP_FILES
+    assert set(files).issuperset(expected_files + CONTENT_FILES), assert_msg
+
+    # restore initial backup and check system health
+    result = sat_maintain.cli.Restore.run(
+        backup_dir=init_backup_dir,
+        options={'assumeyes': True},
+    )
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+    result = sat_maintain.cli.Health.check(
+        options={'whitelist': 'foreman-tasks-not-paused', 'assumeyes': True}
+    )
+    logger.info(result.stdout)
+    assert result.status == 0
+    # assert 'FAIL' not in result.stdout
```

Comment:
should this be commented out?

Reply:
Well, this may be for further discussion, let me explain:
In health check on 7.0 I saw that the ping call failed after restore, so I added `'assumeyes': True` on L222 to allow the check to restart services and commented out the assert (because `FAIL` appeared in stdout once, though the final result after service restart was 0).
So, my question is what should we do here - request NO ping failures after restore and fail here immediately (then un-comment the assert and/or remove the `assumeyes`), or not to be so strict and allow service restart and check the status only after that (then remove the commented line).

Any ideas?

Reply:
I checked how the tool works and I think the status check is sufficient here, because it basically says the fail situation was remediated. You could run the command again to get clean output, but I don't think its necessary